### PR TITLE
chore(TBD-13325):CVE on protobuf-java-2.5.0

### PR DESCRIPTION
**Pull request by lbourgeois**
Cleaning done in local provider for distros older than :
- EMR 45
- CDH 58
- HDP 250
- MAPR 50

Cleaning done also for orphans libraryNeeded definition

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 1]
```diff
@@ -84,7469 +77,420 @@
...
-
         <libraryNeeded
-            context="plugin:org.talend.libraries.hadoop.hdp.2.3"
-            id="hive-serde-HDP_2_3"
-            name="hive-serde-1.2.1.2.3.2.0-2950.jar" mvn_uri="mvn:org.talend.libraries/hive-serde-1.2.1.2.3.2.0-2950/6.1.0"
-            uripath="platform:/plugin/org.talend.libraries.hadoop.hdp.2.3/lib/hive-serde-1.2.1.2.3.2.0-2950.jar">
+            id="commons-codec-1.4.jar"
+            name="commons-codec-1.4.jar" mvn_uri="mvn:commons-codec/commons-codec/1.4"
+            context="plugin:org.talend.libraries.apache.common"
+            uripath="platform:/plugin/org.talend.libraries.apache.common/lib/commons-codec-1.4.jar">
```
Comment (sponomarova) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml]: wondering if any purpose to have older lib for commons-codec? 
I saw we already have 1.14.

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 2]
```diff
@@ -84,7469 +77,420 @@
...
+            context="plugin:org.talend.libraries.hadoop.hdp.2.2"
+            uripath="platform:/plugin/org.talend.libraries.hadoop.hdp.2.2/lib/tezServer/htrace-core-3.0.4.jar">
         </libraryNeeded>
-
         <libraryNeeded
-            context="plugin:org.talend.libraries.hadoop.hdp.2.3"
-            id="talend-spark-assembly-HDP_2_3"
-            name="talend-spark-assembly-1.4.1.2.3.2.0.jar" 
-            mvn_uri="mvn:org.talend.libraries/talend-spark-assembly-1.4.1.2.3.2.0/6.1.0">
+            id="httpclient-4.2.5.jar"
```
Comment (sponomarova) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml]: question about version.

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 3]
```diff
@@ -84,7469 +77,420 @@
...
+            uripath="platform:/plugin/org.talend.libraries.hadoop.hdp.2.2/lib/tezServer/slf4j-api-1.7.5.jar">
         </libraryNeeded>
-
         <libraryNeeded
-            context="plugin:org.talend.libraries.hadoop.hdp.2.3"
-            id="tez-mapreduce-HDP_2_3"
-            name="tez-mapreduce-0.7.0.2.3.2.0-2950.jar" mvn_uri="mvn:org.talend.libraries/tez-mapreduce-0.7.0.2.3.2.0-2950/6.1.0"
-            uripath="platform:/plugin/org.talend.libraries.hadoop.hdp.2.3/lib/tez-mapreduce-0.7.0.2.3.2.0-2950.jar">
+            id="slf4j-log4j12-1.7.5.jar"
+            name="slf4j-log4j12-1.7.5.jar" mvn_uri="mvn:org.talend.libraries/slf4j-log4j12-1.7.5/6.0.0"
```
Comment (sponomarova) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml]: question about version.
I saw 1.7.25 even.

Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.localprovider/plugin.xml]: You're right that's not clear. I will do more manual cleaning keeping only the libs defined in module groups which left

**Review by AlixMetivier — APPROVED** [review]

**Pull request merged**